src/Avalonia.Controls/Primitives/TextSelectionCanvas.cs

```diff
@@ -234,12 +231,11 @@ public void MoveHandlesToSelection()
                 var start = Math.Min(selectionStart, selectionEnd);
                 var length = Math.Max(selectionStart, selectionEnd) - start;
 
-                var rects = _presenter.TextLayout.HitTestTextRange(start, length);
 
-                if (rects.Any())
+                if (_presenter.TextLayout.HitTestTextRange(start, length) is IReadOnlyList<Rect> { Count: > 0 } rects)
```

Comment:
That's rather problematic, and I don't think should be accepted. As it makes this control to rely on internal implementation of HitTestTextRange.
`HitTestTextRange` can return an enumerable at any point of time, and it would be really hard to find this problem otherwise.